src/base.ts

```diff
@@ -0,0 +1,240 @@
+                                .then(resolve)
+                                .catch(reject);
+                        }, backoffDelayMs);
+                    } else {
+                        resp.json()
+                            .then(body => {
+                                const err =
+                                    that._checkStatusForError(resp.status, body) ||
+                                    _getErrorForNonObjectBody(resp.status, body);
+
+                                if (err) {
+                                    reject(err);
+                                } else {
+                                    resolve({
+                                        statusCode: resp.status,
+                                        headers: resp.headers,
+                                        body,
+                                    });
+                                }
+                            })
+                            .catch(() => {
+                                const err = _getErrorForNonObjectBody(resp.status);
+                                reject(err);
+                            });
+                    }
+                })
+                .catch(err => {
+                    clearTimeout(timeout);
+                    err = new AirtableError('CONNECTION_ERROR', err.message, null);
+                    reject(err);
+                });
+        });
+    }
+
+    /**
+     * @deprecated This method is deprecated.
+     */
+    runAction(method, path, queryParams, bodyData, callback) {
+        runAction(this, method, path, queryParams, bodyData, callback, 0);
+    }
+
+    _getRequestHeaders(headers) {
+        const result = new HttpHeaders();
+
+        result.set('Authorization', `Bearer ${this._airtable._apiKey}`);
+        result.set('User-Agent', userAgent);
+        result.set('Content-Type', 'application/json');
+        forEach(headers, (headerValue, headerKey) => {
+            result.set(headerKey, headerValue);
+        });
+
+        return result.toJSON();
+    }
+
+    _checkStatusForError(statusCode, body) {
+        if (statusCode === 401) {
+            return new AirtableError(
+                'AUTHENTICATION_REQUIRED',
+                'You should provide valid api key to perform this operation',
+                statusCode
+            );
+        } else if (statusCode === 403) {
+            return new AirtableError(
+                'NOT_AUTHORIZED',
+                'You are not authorized to perform this operation',
+                statusCode
+            );
+        } else if (statusCode === 404) {
+            return (() => {
+                const message =
+                    body && body.error && body.error.message
+                        ? body.error.message
+                        : 'Could not find what you are looking for';
+                return new AirtableError('NOT_FOUND', message, statusCode);
+            })();
```

Comment:
Ah. It makes sense if you look at the old version in `lib/base.js`. I'm guessing it's a legacy workaround to get around var hoisting and a linting error for that.

I'm not sure I want to change it so to keep from changing how the code executes. I used `lebab` to do safe translation to js modules and left it's other changes since they should be safe and not change execution. Guess I trust lebab more on that than myself.